ddtrace/appsec/_iast/processor.py

```diff
@@ -63,12 +59,13 @@ def on_span_finish(self, span: Span):
         if span.span_type != SpanTypes.WEB:
             return
 
+        from ._taint_tracking import reset_context  # noqa: F401
+
         if not core.get_item(IAST.REQUEST_IAST_ENABLED, span=span):
             span.set_metric(IAST.ENABLED, 0.0)
+            reset_context()
```

Comment:
Question, do we need this reset context? It's still called on line 81

If we need it regardless, I'd move it to line 63 and remove it from here and 81

Reply:
No, I can't move it. we need in line 66 because a early return. reset_context must called  at 81 because we still need context in `build_and_scrub_value_parts`, `_set_metric_iast_request_tainted` and `_set_span_tag_iast_executed_sink`